system-helper/eos-event-recorder-system-helper.c

```diff
@@ -0,0 +1,195 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+
+/* Copyright 2014 - 2017 Endless Mobile, Inc. */
+
+/*
+ * This file is part of eos-event-recorder-daemon.
+ *
+ * eos-event-recorder-daemon is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-event-recorder-daemon is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-event-recorder-daemon.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <gio/gio.h>
+#include <glib.h>
+#include <glib-unix.h>
+#include <polkit/polkit.h>
+
+#include <uuid/uuid.h>
+
+#include <shared/metrics-util.h>
+
+#include "emer-metrics-system-helper.h"
+
+#define TRACKING_ID_OVERRIDE SYSCONFDIR "/eos-metrics-event-recorder/machine-id-override"
```

Comment:
A better choice of path would be `/etc/metrics/machine-id-override`. The daemon already writes a file to this directory ([`cache-size.conf`](https://github.com/endlessm/eos-event-recorder-daemon/blob/master/daemon/emer-cache-size-provider.c#L39); `eos-metrics-permissions.conf` is also in that directory but for reasons that elude me it is edited by external scripts and re-read by the daemon every time its contents are needed). The `metrics` user owns this directory so there is no need for a privileged helper.

Reply:
Ah of course! That would allow me to drop the system helper too...